state/protocol/protocol_state/state/mutable_protocol_state_test.go

```diff
@@ -398,11 +397,10 @@ func (s *StateMutatorSuite) Test_InvalidParent() {
 	unknownParent := unittest.IdentifierFixture()
 	s.protocolKVStoreDB.On("ByBlockID", unknownParent).Return(nil, storage.ErrNotFound)
 
-	deferredDBOps := deferred.NewDeferredBlockPersist()
-	_, err := s.mutableState.EvolveState(deferredDBOps, unknownParent, s.candidate.View, []*flow.Seal{})
+	_, deferredDBOps, err := s.mutableState.EvolveState(unknownParent, s.candidate.View, []*flow.Seal{})
 	require.Error(s.T(), err)
 	require.False(s.T(), protocol.IsInvalidServiceEventError(err))
-	require.True(s.T(), deferredDBOps.IsEmpty())
+	require.Nil(s.T(), deferredDBOps)
```

Comment:
In case of error or exception, the `deferredDBOps` pointer is by convention `nil`